pallets/loans/src/functions.rs

```diff
@@ -143,32 +145,36 @@ impl<T: Config> Pallet<T> {
 		NextLoanId::<T>::mutate(pool_id, |loan_id| *loan_id = next_loan_id);
 
 		// create loan
-		Loan::<T>::insert(
+		UnpricedLoans::<T>::insert(
 			pool_id,
 			loan_id,
 			LoanDetails {
 				collateral,
-				status: LoanStatus::Created,
+				schedule,
 			},
 		);
 		Ok(loan_id)
 	}
 
-	pub(crate) fn price_created_loan(
+	pub(crate) fn price_loan(
 		pool_id: PoolIdOf<T>,
 		loan_id: T::LoanId,
-		interest_rate_per_year: T::Rate,
-		loan_type: LoanType<T::Rate, T::Balance>,
-	) -> Result<(u32, T::Rate), DispatchError> {
+		pricing: LoanPricingInput<T::Rate, T::Balance>,
+	) -> Result<(), DispatchError> {
 		let now = Self::now();
-		ensure!(loan_type.is_valid(now), Error::<T>::LoanValueInvalid);
```

Comment:
Moving the `maturity_date` to the creation properties of the loan, means that now somebody can price something with a passed maturity date. Should we check this? or the caller of this method should be aware of it?

Reply:
Good point, we should check this in `create` now. Maybe we should have an `is_valid` method on the `RepaymentSchedule` trait and check this there?

Reply:
Ok, and what would happen in the next scenario:
1. create() -> with a correct maturity_date.
1. Times passes until now > maturity_date
1. Somebody tries to price() over the loan. Should we also check the maturity also here?

In fact, I understand that any create/price/reprice/extend should check this too.